### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by oncleben31**
Contribute to #18 by applying good practices of https://cjolowicz.github.io/posts/hypermodern-python-05-documentation/ for documentation:

- Review of the docstrings
- Use darglint in test suite to validate docstrings against function signatures
- convert CONTRIBUTING and README in reST
- Generate project and code documentation with Sphinx
- Add Nox session to manage the documentation build

### CODE REVIEW

**Review by Quentame — COMMENTED** [review]
Half of the review for now.
Will continue.

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Pour proposer vos idées d'amélioration, ouvrez une `issue <https://github.com/hacf-fr/meteofrance-api/issues>`_
+en utilisant le modèle ``feature request``.
+
+Contributing
+------------
+
+You are all invited to contribute to this project for maintenance or improvement.
+Even if you are not a developer, you can probably help to report some bugs, share
```
Comment (Quentame) [inline CONTRIBUTING.rst:87]:
```suggestion
Even if you are not a developer, you can help to report some bugs, share
```
?

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+Déscription
+^^^^^^^^^^^
+
+Ce package Python permet de gérer la communication avec l'API non publique de
+Météo-France utilisée par les applications moblies officielles.
+
+Le client permet:
+
+* Rechercher des lieux de prévisions.
+* Accéder aux prévisions météorologiques horraires ou quotidiennes.
```
Comment (Quentame) [inline README.rst:60]:
```suggestion
* Accéder aux prévisions météorologiques horaires ou quotidiennes.
```

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+* Accéder aux prévisions de pluie dans l'heure quand disponibles.
+* Accéder aux alertes météo pour chaque département français et l'Andorre. Deux
```
Comment (Quentame) [inline README.rst:62]:
```suggestion
* Accéder aux alertes météo pour chaque département français et d'Andorre. Deux
```

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+^^^^^^^^^^
+
+Les contributions sont les bienvenues. Veuillez consulter les bonnes pratiques
+détaillées dans `CONTRIBUTING.rst`_.
+
+
+For English speaking users
+--------------------------
+
+Descritption
```
Comment (Quentame) [inline README.rst:93]:
```suggestion
Description
```

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+   :alt: Black
+
+You will find English README content in the section `For English speaking users`_.
+
+Vous trouverez le contenu francophone du README dans la section `Pour les francophones`_.
+
+Pour les francophones
+---------------------
+
+Déscription
```
Comment (Quentame) [inline README.rst:51]:
```suggestion
Description
```

File: README.rst [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+
+For English speaking users
+--------------------------
+
+Descritption
+^^^^^^^^^^^^
+
+This Python package manages the communication with the private Météo-France API
+used by the official moblie applications.
```
Comment (Quentame) [inline README.rst:97]:
```suggestion
used by the official mobile applications.
```

File: README.rst [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+Vous trouverez le contenu francophone du README dans la section `Pour les francophones`_.
+
+Pour les francophones
+---------------------
+
+Déscription
+^^^^^^^^^^^
+
+Ce package Python permet de gérer la communication avec l'API non publique de
+Météo-France utilisée par les applications moblies officielles.
```
Comment (Quentame) [inline README.rst:55]:
```suggestion
Météo-France utilisée par les applications mobiles officielles.
```

File: src/meteofrance/client.py [hunk 8]
```diff
@@ -114,15 +159,21 @@ def get_warning_current_phenomenoms(
...
-
-        depth: use 1 with `france` domain to have all sub location phenomenoms.
+        Args:
+            domain: could be `france` or any metropolitan France department numbers on
+                two digits. For some departments you can access an additional bulletin
+                for coastal phenomenoms. To access it add `10` after the domain id
+                (example: `1310`).
+            depth: Optional; To be used with domain = 'france'. With depth = 0 the
+                results will show only natinal sum up of the weather alerts. If
+                depth = 1, you wil have in addtion the buletin for all metropolitan
```
Comment (Quentame) [inline src/meteofrance/client.py:169]:
```suggestion
                depth = 1, you will have in addition, the bulletin for all metropolitan
```

File: src/meteofrance/client.py [hunk 9]
```diff
@@ -149,19 +200,24 @@ def get_warning_current_phenomenoms(
...
-        the next 24 hours, a list of alerts and other metadatas.
-
-        domain: could be `france` or any department numbers on two digits.
-        For some department you can access an additional bulletin for coastal
-        phenomenoms.
-        To access it add `10` after the domain id (example: `1310`).
+        evolution for the next 24 hours, a list of alerts and other metadatas.
 
         with_costal_bulletin: If set to True, you can get the basic bulletin and
         coastal bulletin merged.
```
Comment (Quentame) [inline src/meteofrance/client.py:209]:
```suggestion
```

File: src/meteofrance/helpers.py [hunk 10]
```diff
@@ -46,22 +58,44 @@ def get_phenomenon_name_from_indice(
...
 
 
 def readeable_phenomenoms_dict(
     list_phenomenoms: List[Dict[str, int]], language: str = "fr"
 ) -> Dict[Optional[str], Optional[str]]:
-    """Create a dictionary with human readable keys and values (Helper)."""
+    """Create a dictionary with human readable keys and values (Helper).
+
+    Args:
+        list_phenomenoms: Dictionnary with phenomnom ID and color code of status.
```
Comment (Quentame) [inline src/meteofrance/helpers.py:92]:
```suggestion
        list_phenomenoms: Dictionary with phenomenon ID and color code of status.
```

File: src/meteofrance/helpers.py [hunk 11]
```diff
@@ -46,22 +58,44 @@ def get_phenomenon_name_from_indice(
...
+        language: Optional; If language is equal "fr" (default value) results will
+            be in French. All other value will give results in English.
+
+    Returns:
+        Dictionnary with keys and value human readable.
+    """
     # Init empty dictionnary
```
Comment (Quentame) [inline src/meteofrance/helpers.py:99]:
```suggestion
        Dictionary with keys and value human readable.
    """
    # Init empty dictionary
```

In more places ?

File: src/meteofrance/helpers.py [hunk 12]
```diff
@@ -101,9 +141,17 @@ def sort_places_versus_distance_from_coordinates(
 ) -> List[Place]:
     """Oder list of places according to the distance to a reference coordinates.
 
-    gps_coordinates are in degrees.
     Note: this helper is compensating the bad results of the API. Results in the API
-    are sorted but lot of case identified where it doesn't work (example: Montréal)
+    are generaly sorted, but lot of cases identified where the order is inconsistent
```
Comment (Quentame) [inline src/meteofrance/helpers.py:145]:
```suggestion
    are generally sorted, but lot of cases identified where the order is inconsistent
```

File: src/meteofrance/model/forecast.py [hunk 13]
```diff
@@ -27,10 +27,31 @@ class ForecastData(TypedDict):
...
+
+    Attributes:
+        position: A dictionary with metadata about the position of the forecast place.
+        updated_on: A timestamp as int corresponding to the latest update date.
+        daily_forecast: A list of dictionaries to describe the daily forecast for the
+            next 15 days.
+        forecast: A list of dictionaries to describe the hourly forecast for the next
+            days.
+        probability_forecast: A list of dictionaries to describe the event probability
+            forecast (rain, snwow, freezing) for next 10 days.
```
Comment (Quentame) [inline src/meteofrance/model/forecast.py:40]:
```suggestion
            forecast (rain, snow, freezing) for next 10 days.
```

File: src/meteofrance/model/place.py [hunk 14]
```diff
@@ -36,22 +23,53 @@ class PlaceData(TypedDict):
...
+
+    Attributes:
+        insee: A string corresponding to the INSEE ID of the place.
+        name: Name of the place.
+        lat: A float with the latitude in degree of the place.
+        lon: A float with the longitude in degree of the place
+        country: A string corresponding to the country code of the place.
+        admin: A string with the name of the administrative area ('Département' for
+            France and Region for other countries).
+        admin2: A string correponding to an administrative code ( 'Département' number
```
Comment (Quentame) [inline src/meteofrance/model/place.py:36]:
```suggestion
        admin2: A string correponding to an administrative code ('Département' number
```

File: src/meteofrance/model/place.py [hunk 15]
```diff
@@ -36,22 +23,53 @@ class PlaceData(TypedDict):
...
+        """
         self.raw_data = raw_data
 
     def __repr__(self) -> str:
-        """Return string representation of this class."""
+        """Return string representation of this class.
+
+        Returns:
+            A string to represent the instance of the Place class using the name,
+            country and amdin area  of the location.
```
Comment (Quentame) [inline src/meteofrance/model/place.py:55]:
```suggestion
            country and admin area of the location.
```

File: src/meteofrance/model/rain.py [hunk 16]
```diff
@@ -25,10 +25,23 @@ class RainData(TypedDict):
...
 class Rain:
-    """Class to access the results of 'rain' API command."""
-
-    def __init__(self, raw_data: RainData):
-        """Initialize a Rain object."""
+    """Class to access the results of 'rain' REST API request.
+
+    Attributes:
+        position: A dictionary with metadata about the position of the forecast place.
+        position: A dictionary with metadata about the position of the forecast place.
```
Comment (Quentame) [inline src/meteofrance/model/rain.py:32]:
```suggestion
        updated_on:  A timestamp as int corresponding to the latest update date.
```

File: src/meteofrance/model/warning.py [hunk 17]
```diff
@@ -74,14 +90,26 @@ def phenomenons_max_colors(self) -> List[Dict[str, int]]:
...
         self.raw_data["phenomenons_max_colors"].extend(
             coastal_phenomenoms.phenomenons_max_colors
         )
 
     def get_domain_max_color(self) -> int:
-        """Get the maximum level of alert of a given domain (class helper)."""
+        """Get the maximum level of alert of a given domain (class helper).
+
+        Returns:
+            An integer correspondingt to the status code representing the maximul alert.
```
Comment (Quentame) [inline src/meteofrance/model/warning.py:111]:
```suggestion
            An integer corresponding to the status code representing the maximum alert.
```

File: src/meteofrance/model/warning.py [hunk 18]
```diff
@@ -96,10 +124,29 @@ class Full:
...
     domains.
+
+    Attributes:
+        update_time: A timestamp (as integer) corresponding to the latest update of the
+            pheomenoms.
+        end_validity_time: A timestamp (as integer) corresponding to expiration date of
+            the phenomenoms.
+        domain_id: A string corresponding do the domain ID of the bulletin. Value is
+            'France' or a department number.
+        color_max: An integer reprenting the maximum alert level in the domain.
```
Comment (Quentame) [inline src/meteofrance/model/warning.py:135]:
```suggestion
        color_max: An integer representing the maximum alert level in the domain.
```

File: src/meteofrance/model/warning.py [hunk 19]
```diff
@@ -133,7 +180,15 @@ def phenomenons_items(self) -> List[Dict[str, int]]:
         return self.raw_data["phenomenons_items"]
 
     def merge_with_coastal_phenomenons(self, coastal_phenomenoms: "Full") -> None:
-        """Merge the classical phenomenoms bulleting with the coastal one."""
+        """Merge the classical phenomenoms bulleting with the coastal one.
```
Comment (Quentame) [inline src/meteofrance/model/warning.py:183]:
```suggestion
        """Merge the classical phenomenon bulletin with the coastal one.
```

**Review by Quentame — CHANGES_REQUESTED** [review]
Some words to correct

File: README.rst (hunk 4 above)
Comment (oncleben31) [reply README.rst]: Done

**Comment by oncleben31** [comment]
All comment implemented. A big thank you @Quentame

**Comment by oncleben31** [comment]
PR rebased with master. All comments implemented. All checks OK. LGTM

**Pull request merged**